pkg/rbd/rbd_attach.go

```diff
@@ -218,6 +221,13 @@ func createPath(volOpt *rbdVolume, cr *util.Credentials) (string, error) {
 	// Execute map
 	output, err := execCommand(rbd, mapOptions)
 	if err != nil {
+		// unmap rbd image if connection timeout
+		if strings.Contains(err.Error(), rbdMapConnectionTimeout) {
+			err = detachRBDImageOrDeviceSpec(imagePath, true, isNbd)
```

Comment:
You're overwriting the previous return value of `err` here. The `klog.Warningf()` on line 231 does not print the right error anymore. We need to log the original error too.

Reply:
@nixpanic if its a timeout, thats the error message or error, Isnt it ?

Reply:
yes, nice catch!

Reply:
ah.. we are using the `err` later, so its good to be on different var. thanks.

Reply:
Using a different variable is ok, or you could have moved the log from line 231 further up. In case mapping fails, and unmapping too, there will be a log about unmapping first, and the error about mapping 2nd. That is just cosmetics though, so not much of an issue